### PR TITLE
Converted layout page to react

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/3673

## Description of Changes
- Converted legacy mithril layout component to react
- Per discussion with @jnaviask, Removed logic to initNewTokenChain and generate a new community when the provided address and scope are eth based
- Removed stale survey code
- Added init steps taken from [this](https://docs.google.com/document/d/12GAC5JVI7mA1Wkq6bD3xBoz1OB7DngF7_9NABPCVRyc/edit#) doc as comments for related code blocks - TY @jnaviask 

## Test Plan
All new layout comp function should work as before
- Switching b/w communities should work
- Switching b/w a community and a non-community page (like the landing page) should work
- Defer chain logic should work as it did before

## Deployment Plan
N/A

## Other Considerations
N/A

### CODE REVIEW

**Comment by mzparacha** [comment]
Committed in multiple steps for easily navigating through the changes, maybe we could squash it into a single commit

File: packages/commonwealth/client/scripts/views/layout.tsx [hunk 1]
```diff
@@ -48,125 +64,169 @@ const shouldDeferChain = ({ deferChain }: ShouldDeferChainAttrs) => {
...
+  // immediately (before selectChain resolves).
+  // -
+  // IFB 6: If deferChain is false on the page we’re routing to, but we
+  // have loaded with isChainDeferred=true (previously from step 5),
+  // then call initChain and render a LoadingLayout immediately.
+  // -
+  // IFB 7: If scope is not defined (and we are not on a custom domain),
+  // deinitialize whatever chain is loaded by calling deinitChainOrCommunity,
+  // then set loadingScope to null. Render a LoadingLayout immediately.
+  if (
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/layout.tsx:193]: Will this case render properly if e.g. isLoading changes?

**Review by jnaviask — CHANGES_REQUESTED** [review]
Really great stuff. Added some comments for cleanup. Have not QA'd yet.

File: packages/commonwealth/client/scripts/views/layout.tsx [hunk 2]
```diff
@@ -48,125 +64,169 @@ const shouldDeferChain = ({ deferChain }: ShouldDeferChainAttrs) => {
...
+      app.chain &&
+      app.chain.network
+    ) {
+      setIsLoading(true);
+      deinitChainOrCommunity().finally(() => {
+        setScopeToLoad(null);
+        setIsLoading(false);
+      });
     }
+  }, [selectedScope]);
```
[the remark this answers is not in the document]
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/layout.tsx:165]: useEffect only re-runs when any props or state value in dependency array changes. So passing `app.chain` won't cause it to re-run.

Some sample code to test this behaviour
```
import { useEffect } from "react";

let val = 1;

setInterval(() => {
  val += 1;
  console.log("value changed");
}, 1000);

function Test() {
  useEffect(() => {
    console.log("re-rendering");
  }, [val]);

  return <></>;
}

export default Test;

```

File: packages/commonwealth/client/scripts/views/layout.tsx [hunk 3]
```diff
@@ -48,125 +64,169 @@ const shouldDeferChain = ({ deferChain }: ShouldDeferChainAttrs) => {
...
+ * - IBS 3 is omitted
+ */
+const LayoutComponent = ({
+  // router,
+  children,
+  scope: selectedScope,
+  deferChain: shouldDeferChain,
+}: LayoutAttrs) => {
+  const scopeMatchesChain = app.config.chains.getById(selectedScope);
+  const scopeIsEthereumAddress =
```
[the remark this answers is not in the document]
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/layout.tsx]: Done

File: packages/commonwealth/client/scripts/views/layout.tsx (hunk 1 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/layout.tsx:193]: Yes, it has less to do with isLoading and more with `scope` related stuff

Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/layout.tsx:193]: I verified by first going to `/dydx` directly and then using click to go back to `/for-you` page and the opposite

File: packages/commonwealth/client/scripts/views/layout.tsx [hunk 4]
```diff
@@ -48,125 +64,169 @@ const shouldDeferChain = ({ deferChain }: ShouldDeferChainAttrs) => {
...
+  // -
+  // IFB 7: If scope is not defined (and we are not on a custom domain),
+  // deinitialize whatever chain is loaded by calling deinitChainOrCommunity,
+  // then set loadingScope to null. Render a LoadingLayout immediately.
+  if (
+    isLoading ||
+    !app.loginStatusLoaded() ||
+    // Important: render loading state immediately for IFB 5, 6 and 7
+    // For IFB 5
+    (selectedScope &&
```
[the remark this answers is not in the document]
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/layout.tsx:198]: possible, doing it now

[the remark this answers is not in the document]
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/layout.tsx]: Done [0bc267a](https://github.com/hicommonwealth/commonwealth/pull/3685/commits/0bc267a83cdd4d2ed12813a73c4a78e51031fcf8)

File: packages/commonwealth/client/scripts/views/layout.tsx [hunk 5]
```diff
@@ -48,125 +64,169 @@ const shouldDeferChain = ({ deferChain }: ShouldDeferChainAttrs) => {
...
+      // time (and we are not loading another community at the same time,
+      // via this.loadingScope), set this.loadingScope to the provided scope,
+      // and then call selectChain, passing deferChain through. If deferChain
+      // is false once selectChain returns, call initChain. Render a LoadingLayout
+      // immediately (before selectChain resolves).
+      if (selectedScope !== app.activeChainId() && scopeMatchesChain) {
+        setIsLoading(true);
+        setScopeToLoad(selectedScope);
+        setIsChainDeferred(true);
+        selectChain(scopeMatchesChain, shouldDeferChain).then((response) => {
```
[the remark this answers is not in the document]
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/layout.tsx]: Done

File: packages/commonwealth/client/scripts/views/layout.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jnaviask) [reply packages/commonwealth/client/scripts/views/layout.tsx]: Great, thanks for the clarification.

**Review by jnaviask — APPROVED** [review]
LGTM

**Review by jnaviask — APPROVED** [review]
Bug I caught is fixed by #3705 -- this PR can go out after that.

**Pull request merged**